openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -55,18 +58,16 @@ def __init__(self, course_key, request, username=''):
             username or request.user.username,
             course_key,
         )
-        self.effective_user = self.overview.effective_user
         self.original_user_is_staff = has_access(request.user, 'staff', self.overview).has_access
         self.course_key = course_key
         self.enrollment_object = CourseEnrollment.get_enrollment(self.effective_user, self.course_key,
                                                                  select_related=['celebration'])
-        course_masquerade, user = setup_masquerade(
+        self.course_masquerade, self.effective_user = setup_masquerade(
```

Comment:
This seems correct, to use the newly returned masquerade user object as the `effective_user`.

Which I _think_ is what we want; we have the `original_user_is_staff` flag to key on when needed for high-level checks.

I'm not sure why this was done originally (to use the `overview.effective_user`) and the `git blame` isn't obvious (the change came as part of a big refactor) :shrug: